tests/config_test.cc

```diff
@@ -12,11 +12,14 @@
 #include <boost/program_options.hpp>
 
 #include "bootstrap.h"
-#include "config.h"
+
 #include "crypto.h"
 #include "test_utils.h"
 #include "utils.h"
 
+#define private public  // We need access to private members
```

Comment:
Woah, that seems brutal. Is that really the best way to do it? I looked into friending the test class before, but I wasn't very happy with that, so I don't really have anything else to suggest at the moment.

Reply:
Why we worry about that? It is only tests

Reply:
True, but may mean our tests are not testing what we expect them to test in subtle ways. For the record, my previous reference on the subject was here: https://stackoverflow.com/questions/3676664/unit-testing-of-private-methods

The last time this came up for me, I ultimately gave up and made the method I wanted to test public. That was lazy and we ought to decide on a better strategy.